src/Evolution/Systems/GeneralizedHarmonic/ConstraintDamping/TimeDependentTripleGaussian.hpp

```diff
@@ -0,0 +1,203 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <limits>
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/ConstraintDamping/DampingFunction.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+namespace PUP {
+class er;
+}  // namespace PUP
+namespace domain::FunctionsOfTime {
+class FunctionOfTime;
+}  // namespace domain::FunctionsOfTime
+/// \endcond
+
+namespace GeneralizedHarmonic::ConstraintDamping {
+/*!
+ * \brief A sum of three Gaussians plus a constant, where the Gaussian widths
+ * are scaled by a domain::FunctionsOfTime::FunctionOfTime.
+ *
+ * \details The function \f$f\f$ is given by
+ * \f{align}{
+ * f = C + \sum_{\alpha=1}^3
+ * A_\alpha \exp\left(-\frac{(x-(x_0)_\alpha)^2}{w_\alpha^2(t)}\right).
+ * \f}
+ * Input file options are: `Constant` \f$C\f$, `Amplitude[1-3]`
+ * \f$A_\alpha\f$, `Width[1-3]` \f$w_\alpha\f$, `Center[1-3]
+ * `\f$(x_0)_\alpha\f$, and `FunctionOfTimeForScaling`, a string naming a
+ * domain::FunctionsOfTime::FunctionOfTime in the domain::Tags::FunctionsOfTime
+ * that will be passed to the call operator. The function takes input
+ * coordinates \f$x\f$ of type `tnsr::I<T, VolumeDim, Fr>`, where `T` is e.g.
+ * `double` or `DataVector`, `Fr` is a frame (e.g. `Frame::Inertial`), and
+ * `VolumeDim` is the dimension of the spatial volume. The Gaussian widths
+ * \f$w_\alpha\f$ are scaled by the inverse of the value of a scalar
+ * domain::FunctionsOfTime::FunctionOfTime \f$f(t)\f$ named
+ * `FunctionOfTimeForScaling`: \f$w_\alpha(t) = w_\alpha / f(t)\f$.
+ */
+template <size_t VolumeDim, typename Fr>
```

Comment:
why is this templated on VolumeDim and Frame? I don't think we need it for anything besides 3 and Frame::Grid

Reply:
I was trying to preserve flexibility in case in the (far) future we want to explore different dimensions or frames (eg we might someday want to try this in the distorted frame, and perhaps someday there might be a 2D axisymmetric simulation we might try). Also I preferred the consistency of templatimg all damping functions the same way. Finally, this forces the user to expressly choose dim and frame, so they don't run into surprises trying it in a different case than the hard coded one.

If you don't buy any of my reasons, I'm happy to hard code 3, Frame::Grid3.

Reply:
no, since the centers are fixed by input-file parameters, the only frame in which they make sense is Frame::Grid since it is the only frame in which we can guarantee the excision surface remains fixed.  And for 2D axisymmetric simulations, our plan is to handle them  through a cartoon method acting on 3d tensors.   And also since this should only be used on a BinaryCompactObject domain which is a 3D domain.